File: src/main/java/org/example/gogoma/controller/response/UserMarathonSearchResponse.java
package org.example.gogoma.controller.response;

import lombok.Builder;
import lombok.Getter;
import org.example.gogoma.domain.usermarathon.dto.UserMarathonSearchDto;

import java.util.List;

@Getter
@Builder
public class UserMarathonSearchResponse {
    private List<UserMarathonSearchDto> userMarathonSearchDtoList;

    public static UserMarathonSearchResponse of(List<UserMarathonSearchDto> userMarathonSearchDtoList) {
        return UserMarathonSearchResponse.builder()
                .userMarathonSearchDtoList(userMarathonSearchDtoList)
                .build();
    }
}

File: app/src/main/java/com/example/gogoma/data/model/UserMarathonSearchResponse.kt
package com.example.gogoma.data.model

import android.os.Parcelable
import com.example.gogoma.data.dto.UserMarathonSearchDto
import kotlinx.parcelize.Parcelize

@Parcelize
data class UserMarathonSearchResponse(
    val userMarathonSearchDtoList: List<UserMarathonSearchDto>
) : Parcelable
